scipy/sparse/csgraph/_shortest_path.pyx

```diff
@@ -166,12 +166,17 @@ def shortest_path(csgraph, method='auto',
         issparse = isspmatrix(csgraph)
         if issparse:
             Nk = csgraph.nnz
+            edges = csgraph.data
+        elif np.ma.isMaskedArray(csgraph):
+            Nk = csgraph.count()
+            edges = csgraph.compressed()
         else:
+            csgraph[~np.isfinite(csgraph)] = 0
```

Comment:
Modifies input --- you probably can write instead
```
edges = csgraph[np.isfinite(csgraph)]
```
Should you also then write `Nk = edges.size` --- `np.sum(csgraph > 0)` counts also inf entries?

Reply:
While I was trying to make the changes you suggested, I noticed that I forgot to count the negative edges(I counted positive values ​​instead of non-zero ones). I fixed it now and removed the input modification.